integration/src/test/java/dev/ikm/tinkar/integration/search/SearcherIT.java

```diff
@@ -121,4 +126,73 @@ public void searchFromDescendantsOfConceptWithCustomCalculatorIT() throws Except
+        List<PublicId> constraintAncestors = new ArrayList<>();
+        PublicId constraintId = PublicIds.of("7bbd4210-381c-11e7-9598-0800200c9a66");
+        constraintAncestors.add(constraintId);
+        PublicId ancestorId = PublicIds.of("3642d9a3-8e23-5289-836b-366c0b1e2900");
+        List<PublicId> descendants = Searcher.getDescendantsWithConstraint(ancestorId, constraintAncestors);
+        assertEquals(0, descendants.size());
+    }
+
+    @Test
+    public void descendantsWithConstraintTestChildrenEmptyList() throws Exception {
+        List<PublicId> constraintAncestors = new ArrayList<>();
+        PublicId ancestorId = PublicIds.of("f7495b58-6630-3499-a44e-2052b5fcf06c");
+        List<PublicId> descendants = Searcher.getDescendantsWithConstraint(ancestorId, constraintAncestors);
+        assertEquals(8, descendants.size());
+    }
+
+    @Test
+    public void descendantsWithConstraintTestChildrenNonEmptyList() throws Exception {
+        List<PublicId> constraintAncestors = new ArrayList<>();
+        PublicId constraintId = PublicIds.of("f7495b58-6630-3499-a44e-2052b5fcf06c");
+        constraintAncestors.add(constraintId);
+        PublicId ancestorId = PublicIds.of("7bbd4210-381c-11e7-9598-0800200c9a66");
+        List<PublicId> descendants = Searcher.getDescendantsWithConstraint(ancestorId, constraintAncestors);
+        assertEquals(181, descendants.size());
+    }
+
+
+    @Test
+    public void descendantsWithConstraintConstraintNotAConcept() throws Exception {
+        List<PublicId> constraintAncestors = new ArrayList<>();
+        PublicId constraintId = PublicIds.of("9c25d708-b9e9-4e97-b9f2-cba5dc917975");
+        constraintAncestors.add(constraintId);
+        PublicId ancestorId = PublicIds.of("7bbd4210-381c-11e7-9598-0800200c9a66");
+        List<PublicId> descendants = Searcher.getDescendantsWithConstraint(ancestorId, constraintAncestors);
+        assertEquals(181, descendants.size());
+    }
+
+    @Test
+    public void descendantsWithConstraintConstraintDNE() throws Exception {
+        List<PublicId> constraintAncestors = new ArrayList<>();
+        PublicId constraintId = PublicIds.of("abcde-6630-3499-a44e-2052b5fcf06c");
+        constraintAncestors.add(constraintId);
+        PublicId ancestorId = PublicIds.of(UUID.fromString("f7495b58-6630-3499-a44e-2052b5fcf06c"));
+        List<PublicId> descendants = Searcher.getDescendantsWithConstraint(ancestorId, constraintAncestors);
+        assertEquals(8, descendants.size());
+    }
+
+    @Test
+    public void descendantsWithConstraintConstraints() throws Exception {
+        List<PublicId> constraintAncestors = new ArrayList<>();
+        PublicId constraintId = TinkarTerm.ROLE_OPERATOR.publicId();
+        constraintAncestors.add(constraintId);
+        PublicId ancestorId = TinkarTerm.ROLE.publicId();
+        List<PublicId> descendants = Searcher.getDescendantsWithConstraint(ancestorId, constraintAncestors);
+        System.out.println(descendants);
```

Comment:
This is probably left over from debugging the test. You can delete it.

Reply:
@stenkarl, you are correct that this was used for debugging, but since it is still a valid test shouldn't we still keep it?

Reply:
Ah, not the entire test, I meant the System.out.println.